diagnostic.js

```diff
@@ -29,26 +29,72 @@ mongoose.Promise = global.Promise;
 mongoose.connect('mongodb://localhost/mongoose-crud');
 
 /// ADD YOUR CODE BELOW
+const db = mongoose.connection;
+const Movement = require('./models/movement.js')
+const done = function() {  db.close(); };  //close connection
 
-const create = (name, description, startYear, endYear) => {};
+const create = (name, description, startYear, endYear) => {
+  Movement.create({
+    name: name,
+    description: description,
+    "starting year" : startYear,
+    "ending year" : endYear,
+  }).then(function(movement){
+    console.log(movement.toJSON());
+  }).catch(console.error).then(done);
+};
 // Success -> Print new Movement as JSON
 // Failure -> Console.error
 
-const index = () => {};
+const index = () => {
+  let searchParams = {};
+   if(arguments[0] && arguments[1]){
+     let field = arguments[0];
+     let criterion = arguments[1];
+     if (criterion[0] === '/'){  //if criterion is a regular expression.
+       let regex = new RegExp(criterion.slice(1, criterion.length - 1));
+       searchParams[field] = regex;
+     } else {
+       searchParams[field] = criterion;
+     }
+   }
+   Movement.find(searchParams).then(function(movements){
+     movements.forEach(function(movement) {
+       console.log(movement.toJSON());
+     });
+   }).catch(console.error).then(done);
+};
 // Success -> Print all Movements as JSON
 // Failure -> Console.error
 
-const show = (id) => {};
+const show = (id) => {
+  Movement.findById(id).then(function(movement){
+    console.log(movement.toObject());
+  }).catch(console.error).then(done);
+};
```

Comment:
Not finding something is not the same as having an error. As a result, you need to check to see if art exists before printing it out; if it doesn't exist, you were directed to print out "Not Found" and exit (analogous to returning a 404 in Express).